README.md
# GreenSafety - User Platform (YUWEI)

User Platform on T-BOX monitoring.

# Environment

Install Apache

```sh
$ sudo yum install httpd mod_ssl
```

Setup Apache Config

```sh
# [DEV]
$ sudo cp /opt/greensafety/EnvConfig/httpd_dev.conf /etc/httpd/conf/httpd.conf

# [PRD]
$ sudo cp /opt/greensafety/EnvConfig/httpd.conf /etc/httpd/conf/httpd.conf
```

Install PHP

```sh
$ sudo yum install https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
$ sudo yum install http://rpms.remirepo.net/enterprise/remi-release-7.rpm
$ sudo yum install yum-utils
$ sudo yum-config-manager --enable remi-php72
$ sudo yum install php php-mcrypt php-cli php-gd php-curl php-mysql php-ldap php-zip php-fileinfo
$ sudo yum -y install gcc php-pear php-devel
$ sudo yum install openssl openssl-devel
$ sudo pecl install mongodb
$ curl https://packages.microsoft.com/config/rhel/7/prod.repo > /etc/yum.repos.d/mssql-release.repo
$ yum remove unixODBC-utf16 unixODBC-utf16-devel
$ ACCEPT_EULA=Y yum install -y msodbcsql17
$ ACCEPT_EULA=Y yum install -y mssql-tools
$ echo 'export PATH="$PATH:/opt/mssql-tools/bin"' >> ~/.bash_profile
$ echo 'export PATH="$PATH:/opt/mssql-tools/bin"' >> ~/.bashrc
$ source ~/.bashrc
$ yum install -y unixODBC-devel
$ yum install -y centos-release-scl
$ yum install -y devtoolset-7
$ scl enable devtoolset-7 bash
$ yum install -y php-sqlsrv
```

Setup PHP Config

```sh
$ cd /usr/lib64/php/modules/
$ sudo chmod u+rwx,g+x,o+x mongodb.so
$ sudo cp /opt/greensafety/EnvConfig/30-mongodb.ini /etc/php.d/30-mongodb.ini
$ sudo cp /opt/greensafety/EnvConfig/php.ini /etc/php.ini
```

Insatll NodeJS

```sh
$ sudo yum install -y gcc-c++ make
$ sudo curl -sL https://rpm.nodesource.com/setup_11.x | sudo -E bash -
$ sudo yum install nodejs
$ sudo npm install -g npm@latest
```

Setup SElinux

```sh
$ sudo setenforce 1
$ sudo chcon -R -t httpd_sys_rw_content_t /opt/greensafety/logs
$ sudo setsebool -P httpd_can_network_connect_db 1
$ sudo setsebool -P httpd_can_network_memcache 1
$ sudo setsebool -P httpd_can_sendmail 1
$ sudo setsebool -P httpd_can_network_connect 1

#enable write on dir
$ sudo chcon -R -t httpd_sys_rw_content_t dir
```

Setup Firewall (CentOS7)

```sh
$ sudo firewall-cmd --permanent --add-port=80/tcp
$ sudo firewall-cmd --permanent --add-port=443/tcp
$ sudo firewall-cmd --reload
```

# Development

Install Git
```sh
$ sudo yum install git-all
```

Setup Local Git Repository

```sh
$ mkdir /opt/web-repo.git
$ cd web-repo.git/
$ git init --bare
$ sudo cp /opt/greensafety/EnvConfig/post-receive /opt/web-repo.git/hooks/post-receive
```

Set Directory Permission
```sh
$ cd /opt/
$ sudo chown -R root:wheel greensafety
$ sudo chown -R root:wheel web-repo.git
$ sudo chown -R root:wheel greensafety
$ sudo chmod -R a+rwx web-repo.git
```

Setup Git SSH

```sh
$ mkdir .ssh
$ touch .ssh/authorized_keys
$ cat id_rsa.pub >> .ssh/authorized_keys
```

Development Directory

```
/opt/greensafety/
```

Install PHP Dependencies

```sh
$ php composer.phar install
```

Package Control

```sh
$ npm
```

Package Directory

```
/opt/greensafety/node_modules/
```

Install Node Modules

```sh
$ npm install
```

Install apiDoc
```sh
$ npm install apidoc -g
```

Install JSDoc
```sh
$ npm install jsdoc jsdoc-mermaid -g
```

Dependancy Rewrite

```sh
# /opt/greensafety/node_modules ==> /opt/greensafety/html/scirpts/
$ ln -s /opt/greensafety/node_modules /opt/greensafety/html/scirpts
```

JSDoc Custom Template

```sh
$ sudo cp /opt/greensafety/EnvConfig/layout.tmpl /opt/greensafety/node_modules/better-docs/tmpl/layout.tmpl
```

Web Content Directory

```
/opt/greensafety/html/
```

Reminder (HTML)

+ typeahead input must contains **id** start with **typeahead#** (bug fixed required)
```
<input id="typeahead#1">
```

Build Version

```sh
$ sudo npm run build-version [major | minor | patch]
```

Build Package

```sh
$ sudo npm run build-package
```

Update Change Log

```sh
$ sudo npm run changelog [add | update | bugfix | remove] [message]
```

Generate API Documentation

```sh
$ sudo npm run apidoc
```

Generate JS Documentation

```sh
$ sudo npm run doc
```

# Usage

Portal
=> https://:hostname/:companyCode/login

Admin Panel
=> https://:hostname/tools/adminpanel/login
=> Username: admin
=> Password: 4V8JB02qy7niHg7p

Autologin Tool
=> https://:hostname/tools/autologin/:companyCode/:Credential

Autologin Tool Credential Generate

```js
btoa(encodeURI(usr=username&pwd=password))
```

# Documentation

API Documentation
=> documentation/api/index.html
=> http://:hostname/tools/documentation/api  [DEV]

JS Documentation
=> documentation/web/index.html
=> http://:hostname/tools/documentation/web  [DEV]

# Techology

* [Apache] - An effort to develop and maintain an open-source HTTP server for modern operating systems including UNIX and Windows
* [PHP] - A popular general-purpose scripting language that is especially suited to web development
* [NodeJS] - A JavaScript runtime built on Chrome's V8 JavaScript engine
* [PHP Slim Framework] - a PHP micro framework that helps you quickly write simple yet powerful web applications and APIs
* [AngularJS] - HTML enhanced for web apps
* [AngularUI Router] - The de-facto solution to flexible routing in angular
* [Angular Chart] - Reactive, responsive, beautiful charts for AngularJS based on Chart.js
* [Angular Dynamic Locale] - Module to be able to change the locale at an angularjs application
* [Angular-filter] - Bunch of useful filters for AngularJS (with no external dependencies!)
* [AngularJS Material] - A set of reusable, well-tested, and accessible UI components based on Material Design
* [Smart-Table] - A table module for angular js
* [Angular-Translate] - i18n for your Angular apps, made easy
* [UI Bootstrap] - Bootstrap components written in pure AngularJS by the AngularUI Team
* [AngularJS Dropdown Multiselect] - A Bootstrap Dropdown with the power of AngularJS directives
* [Angular Gauge] - A highly customizable gauge directive for Angular JS apps and dashboards
* [Bootstrap] - An open source toolkit for developing with HTML, CSS, and JS
* [ES6 Promise] - A polyfill of the ES6 Promise
* [Font Awesome] - The iconic font and CSS toolkit
* [jQuery] - A fast, small, and feature-rich JavaScript library
* [lrStickyHeader] - Make table headers sticky
* [mw-datepicker-range] - An AngularJS module that extend the ui-bootstrap's datepicker directive to allow a selection of a range
* [ngCsv] - An AngularJS simple directive that turns arrays and objects into downloadable CSV files
* [ng-error] - Angular ngError directive
* [Ng-Idle] - An AngularJS module for detecting and responding to idle users
* [Google Maps AngularJS Directive] - The Simplest AngularJS Google Maps V3 Directive
* [OrgChart] - Organization Chart Plugin
* [stStickyHeader] - Sticky table directive for smart-table
* [SweetAlert2] - A beautiful, responsive, customizable and accessible (WAI-ARIA) replacement for JavaScript's popup boxes.
* [Baidu-map For AngularJS] - A baidu-map directive for AngularJS
* [angular-block-ui] - AngularJS Block UI Module

License
----

**GreenSafety Technology Limited**

[//]: #

   [Apache]: <https://httpd.apache.org/>
   [PHP]: <http://www.php.net/>
   [NodeJS]: <https://nodejs.org/en/>
   [PHP Slim Framework]: <http://www.slimframework.com/>
   [AngularJS]: <http://angularjs.org>
   [AngularUI Router]: <https://ui-router.github.io/>
   [Angular Chart]: <http://jtblin.github.io/angular-chart.js/>
   [Angular Dynamic Locale]: <http://lgalfaso.github.io/angular-dynamic-locale/>
   [Angular-filter]: <https://github.com/a8m/angular-filter>
   [AngularJS Material]: <https://material.angularjs.org/latest/>
   [Smart-Table]: <http://lorenzofox3.github.io/smart-table-website/>
   [Angular-Translate]: <https://angular-translate.github.io/>
   [UI Bootstrap]: <https://angular-ui.github.io/bootstrap/>
   [AngularJS Dropdown Multiselect]: <http://dotansimha.github.io/angularjs-dropdown-multiselect/docs/#/main>
   [Angular Gauge]: <https://ashish-chopra.github.io/angular-gauge/>
   [Bootstrap]: <http://getbootstrap.com/?>
   [ES6 Promise]: <https://github.com/stefanpenner/es6-promise#readme>
   [Font Awesome]: <https://fontawesome.com/v4.7.0/>
   [jQuery]: <https://jquery.com/>
   [lrStickyHeader]: <https://github.com/lorenzofox3/lrStickyHeader>
   [mw-datepicker-range]: <https://github.com/MrWook/mw-datepicker-range>
   [ngCsv]: <https://github.com/asafdav/ng-csv>
   [ng-error]: <https://github.com/andrefarzat/ng-error>
   [Ng-Idle]: <http://hackedbychinese.github.io/ng-idle/>
   [Google Maps AngularJS Directive]: <https://ngmap.github.io/>
   [OrgChart]: <https://github.com/dabeng/OrgChart>
   [stStickyHeader]: <https://github.com/lorenzofox3/stStickyHeader>
   [SweetAlert2]: <https://sweetalert2.github.io/>
   [Baidu-map For AngularJS]: <https://github.com/leftstick/BaiduMapForAngularJS>
   [angular-block-ui]: <https://github.com/McNull/angular-block-ui>

CHANGELOG.md
Release Notes
======

# 5.0.1-27 [2019-11-28]


# 5.0.1-27 [2019-09-27]


# 5.0.1-26 [2019-09-24]


# 5.0.1-25 [2019-09-18]


# 5.0.1-24 [2019-09-09]


# 5.0.1-23 [2019-09-09]


# 5.0.1-22 [2019-09-09]


# 5.0.1-21 [2019-09-09]


# 5.0.1-20 [2019-09-09]


# 5.0.1-19 [2019-09-09]


# 5.0.1-18 [2019-09-09]


# 5.0.1-17 [2019-08-22]


# 5.0.1-16 [2019-08-19]


# 5.0.1-15 [2019-08-15]


# 5.0.1-14 [2019-08-15]


# 5.0.1-13 [2019-08-13]


# 5.0.1-12 [2019-08-08]


# 5.0.1-11 [2019-08-08]


# 5.0.1-10 [2019-08-08]


# 5.0.1-9 [2019-08-08]


# 5.0.1-8 [2019-08-07]


# 5.0.1-7 [2019-08-05]


# 5.0.1-6 [2019-08-02]


# 5.0.1-5 [2019-08-02]


# 5.0.1-4 [2019-08-01]


# 5.0.1-3 [2019-08-01]


# 5.0.1-2 [2019-08-01]


# 5.0.1-1 [2019-07-01]

  * [ADDED] New database mapping
  * [ADDED] Modification for MSS2

html/api/config/setup.php
<?php

/**
 * @Author: Zilvia Kam
 * @Date:   2019-04-04 17:39:21
 * @Last Modified by:   Zilvia Kam
 * @Last Modified time: 2019-09-27 17:24:29
 */

// Slim Framework setting
$config['displayErrorDetails'] = true;
$config['addContentLengthHeader'] = true;
$config['determineRouteBeforeAppMiddleware'] = true;

// Monolog setting
$config['logger']['name'] = 'gst-slim-php';
$config['logger']['level'] = Monolog\Logger::DEBUG;
$config['logger']['path'] = __DIR__ . '/../../../logs/app';

// Application setting
$config['version']['path'] = __DIR__ . '/../../../package.json';

// Host setting
$config['data']['webserver'] = 'http://10.33.22.47';
$config['data']['prd'] = 'http://223.197.178.21';

// Database(SQL Server) setting
$config['db']['host'] = '10.33.22.27';
$config['db']['hostprd'] = '10.33.22.27';
$config['db']['port'] = '1433';
$config['db']['user'] = 'ui';
$config['db']['charset'] = 'UTF-8';
$config['db']['pass']   = 'RDcP9UnXfDmPbwJd';
$config['db']['dbname'] = 'frontend_dev';
$config['db']['prepOpts'] = array(PDO::ATTR_EMULATE_PREPARES => true);

// YUWEI API setting
$config['api']['yuweipath']['host'] = 'http://10.33.22.26:8038';
$config['api']['yuweipath']['hostprd'] = 'http://10.33.22.26:8038';
$config['api']['yuweipath']['addDevice'] = '/api/vehicle/update?';
$config['api']['yuweipath']['editDevice'] = '/api/vehicle/update?';
$config['api']['yuweipath']['deleteDevice'] = '/api/vehicle/remove?';
$config['api']['yuweipath']['addDriver'] = '/api/driverInfo/update?';
$config['api']['yuweipath']['deleteDriver'] = '/api/driverInfo/remove?';
$config['api']['yuweiToken'] = '2935c51f26f473e0';
$config['api']['yuweiParam']['addDevice'] = array('token' => '','vehicleNo' => '','vehicleColor' => '黃色','termNo'=> '','code' => '','gas' => 810000.0,'func' => 16,'photoParam' => 4,'videoParam' => 8,'videoChName' => '{"CH1":"CH1","CH2":"CH2","CH3":"CH3","CH4":"CH4","CH5":"CH5,"CH6":"CH6,"CH7":"CH7,"CH8":"CH8"}','videoRelationCh' => '1,2,3,4,5,6,7,8','watchVideoByFlameout' => 0,'opType' => 0);
$config['api']['yuweiParam']['deleteDevice'] = array('token' => '','vehicleNo' => '');
$config['api']['yuweiParam']['addDriver'] = array('token' => '','companyNo'=>'117440513','driverCode'=>'','driverWorkNumber'=>'','driverName'=>'','sex'=>'男','driverLicence'=>'','identityCard'=>'','password'=>'0','licenseAgency'=>'GST','opType'=>1);
$config['api']['yuweiParam']['deleteDriver'] = array('token' => '', 'driverCode'=>'');
$config['api']['yuweipath']['livehost'] = 'http://10.33.22.26:3003';
$config['api']['yuweipath']['livehostprd'] = 'http://223.197.178.21:3003';
$config['api']['yuweipath']['liveMonitor'] = '/mydemo9/index.html?';

// Google Map API setting
$config['api']['googlepath'] = 'https://roads.googleapis.com/v1/snapToRoads?';
$config['api']['googleKey'] = 'AIzaSyBFFZM0aHlgzT-aL8qXuPT7deIEdDEPqLE';

// Baidu Map API setting
$config['api']['baiduKey'] = '9BDu0lFG2GuMunkgOvDnfYXpf8oGNtF2';
$config['api']['baidupath']['geoconv'] = 'https://api.map.baidu.com/geoconv/v1/?';
$config['api']['baidupath']['trackrectify'] = 'https://api.map.baidu.com/rectify/v1/track?';

// Encryption setting
$config['encrypt']['cipher'] = 'AES-256-CBC';
$config['encrypt']['key'] = base64_encode('GREENSAFETY');
$config['encrypt']['iv'] = '55d0b25e6d28e0e0';

// PHPMailer setting
$config['mailer']['CharSet'] = 'UTF-8';
$config['mailer']['SMTPDebug'] = 2;
$config['mailer']['host'] = 'ud.1025.hk';
$config['mailer']['SMTPAuth'] = true;
$config['mailer']['username'] = 'system@greensafety.com.hk';
$config['mailer']['password'] = 'Abc_1234';
$config['mailer']['SMTPSecure'] = 'tls';
$config['mailer']['port'] = 1025;
$config['mailer']['sender'] = 'no-reply@greensafety.com.hk';
$config['mailer']['SMTPoption'] = array('ssl' => array('verify_peer' => false,'verify_peer_name' => false,'allow_self_signed' => true));
$config['mailer']['Debugoutput'] = 'error_log';

// FTP setting (video)
$config['ftp']['host'] = '10.33.22.26';
$config['ftp']['hostprd'] = '10.33.22.26';
$config['ftp']['port'] = '21';
$config['ftp']['username'] = 'yuweiftp';
$config['ftp']['password'] = 'eUckLCDF8TUCdDuW';

/**
 * Common variable setting
 */
// Enabled Function
$config['data']['userFunc'] = ['LIVE_LOCATION','LIVE_MONITOR','TRIP_DETAIL','TRIP_MAINT', 'TRIP_DRV_IMPORT', 'WARNING_EXPORT', 'VIDEO_BATCH_DOWNLOAD', 'COMPANY_PROFILE','VEHICLE_GRP_PROFILE','VEHICLE_PROFILE','DRIVER_GRP_PROFILE','DRIVER_PROFILE','VEHICLE_GRP_MAINT', 'VEHICLE_MAINT','DRIVER_MAINT','DRIVER_GRP_MAINT','DEVICE_MAINT','USER_MAINT','USER_ROLE_MAINT','ORG_CHART_MAINT','MAIL_NOTIFY_MAINT','ROSTER_MAINT'];
// Warning export headeer setting
$config['data']['exportWarnHeader']['en'] = ['Start Time','End Time','Warning Type','Latitude (WGS84)','Longitude (WGS84)','Duration (s)','Start Speed (kph)','End Speed (kph)','Top Speed (kph)','Rate Of Change (kph/s)','HW (s)','Nearest HW (s)','Driver Action (L: Left Indicator; R: Right Indicator; B: Brakes)','Video Status','Video ID','Dangerous Video (Y: Yes; N: No)','Vehicle','Vehicle Department','Vehicle Group','Vehicle Brand','Vehicle Model','Vehicle Type','Vehicle Run Distance-Within Date Range (KM)','Vehicle Total Run Distance (KM)','Driver','Driver Department','Driver Group'];
$config['data']['exportWarnHeader']['zh-tw'] = ['開始時間','結束時間','警報類型','緯度 (WGS84)','經度 (WGS84)','持續時間 (s)','開始速度 (kph)','結束速度 (kph)','最高速度 (kph)','速度變化 (kph/s)','前車距離 (s)','最近前車距離 (s)','司機行為 (L: 亮起左轉指示燈; R: 亮起右轉指示燈; B: 剎車)','視頻狀態','視頻ID','危險駕駛行為影片 (Y: 是; N: 否)','車輛','車輛部門','車輛分組','車輛品牌','車輛型號','車輛類型','車輛行走里數-日期範圍內 (KM)','車輛總行走里數 (KM)','司機','司機部門','司機分組'];
$config['data']['exportWarnHeader']['zh-cn'] = ['开始时间','结束时间','警报类型','纬度 (WGS84)','经度 (WGS84)','持续时间 (s)','开始速度 (kph)','结束速度 (kph)','最高速度 (kph)','速度变化 (kph/s)','前车距离 (s)','最近前车距离 (s)','司机行为 (L: 亮起左转指示灯; R: 亮起右转指示灯; B: 刹车)','视频状态','视频ID','危险驾驶行为影片 (Y: 是; N: 否)','车辆','车辆部门','车辆分组','车辆品牌','车辆型号','车辆类型','车辆行走里数-日期范围內 (KM)','车辆总行走里数 (KM)','司机','司机部门','司机分组'];
// Vehicle type mapping - Warning Export
$config['data']['vehType'] = '{"OTHERS_DEFAULT": "Others","_CAR": "Car","_VAN": "Van","MINIBUS": "Mini-Bus","_BUS": "Bus","_TRUCK": "Truck","CONCRETE_TRUCK": "Concrete Truck"}';
// Warning type mapping - Warning Export
$config['data']['warningType'] = '{"FCW_L2": "FCW", "LDW_L2": "LDW", "HMW_L2": "HMW", "AAW_L2": "AAW", "ABW_L2": "ABW", "ATW_L2": "ATW", "FDW_L2": "DWW", "CLW_L2": "CLW", "DTW_L2": "DTW", "NDW": "NDW", "YW_L2": "YW"}';
// Enabled warning type
$config['data']['wt'] = ['FCW_L2','LDW_L2','HMW_L2','AAW_L2','ABW_L2','ATW_L2','FDW_L2','CLW_L2','DTW_L2','NDW','YW_L2'];
// Warning count field setting - Profile
$config['data']['fullWarningField'] = ['FCW','FCW_L1','FCW_L2','LDW','LDW_L1','LDW_L2','HMW','HMW_L1','HMW_L2','PCW','PCW_L1','PCW_L2','ZDW','ZDW_L1','ZDW_L2','SSW','BDW','SPW','OHW','AAW','AAW_L1','AAW_L2','ABW','ABW_L1','ABW_L2','ATW','ATW_L1','ATW_L2','FDW','FDW_L1','FDW_L2','CLW','CLW_L1','CLW_L2','SMW','SMW_L1','SMW_L2','DTW','DTW_L1','DTW_L2','DAW','NDW','YW','YW_L1','YW_L2','FMW','EW','RAW','RAW_L1','RAW_L2','LRAW','LRAW_L1','LRAW_L2','RRAW','RRAW_L1','RRAW_L2','LROD','LROD_L1','LROD_L2','RROD','RROD_L1','RROD_L2'];
$config['data']['fullWarningField_CLP'] = ['FCW','FCW_L1','FCW_L2','LDW','LDW_L1','LDW_L2','HMW','HMW_L1','HMW_L2','PCW','PCW_L1','PCW_L2','ZDW','ZDW_L1','ZDW_L2','SSW','BDW','SPW','OHW','AAW','AAW_L1','AAW_L2','ABW','ABW_L1','ABW_L2','ATW','ATW_L1','ATW_L2','SMW','SMW_L1','SMW_L2','DAW','FMW','EW','RAW','RAW_L1','RAW_L2','LRAW','LRAW_L1','LRAW_L2','RRAW','RRAW_L1','RRAW_L2','LROD','LROD_L1','LROD_L2','RROD','RROD_L1','RROD_L2'];
// Safety score field setting - Profile
$config['data']['fullScoreField'] = ['FCW_score','FCW_L1_score','FCW_L2_score','LDW_score','LDW_L1_score','LDW_L2_score','HMW_score','HMW_L1_score','HMW_L2_score','PCW_score','PCW_L1_score','PCW_L2_score','ZDW_score','ZDW_L1_score','ZDW_L2_score','SSW_score','BDW_score','SPW_score','OHW_score','AAW_score','AAW_L1_score','AAW_L2_score','ABW_score','ABW_L1_score','ABW_L2_score','ATW_score','ATW_L1_score','ATW_L2_score','FDW_score','FDW_L1_score','FDW_L2_score','CLW_score','CLW_L1_score','CLW_L2_score','SMW_score','SMW_L1_score','SMW_L2_score','DTW_score','DTW_L1_score','DTW_L2_score','DAW_score','NDW_score','YW_score','YW_L1_score','YW_L2_score','FMW_score','EW_score','RAW_score','RAW_L1_score','RAW_L2_score','LRAW_score','LRAW_L1_score','LRAW_L2_score','RRAW_score','RRAW_L1_score','RRAW_L2_score','LROD_score','LROD_L1_score','LROD_L2_score','RROD_score','RROD_L1_score','RROD_L2_score'];
$config['data']['fullScoreField_CLP'] = ['FCW_score','FCW_L1_score','FCW_L2_score','LDW_score','LDW_L1_score','LDW_L2_score','HMW_score','HMW_L1_score','HMW_L2_score','PCW_score','PCW_L1_score','PCW_L2_score','ZDW_score','ZDW_L1_score','ZDW_L2_score','SSW_score','BDW_score','SPW_score','OHW_score','AAW_score','AAW_L1_score','AAW_L2_score','ABW_score','ABW_L1_score','ABW_L2_score','ATW_score','ATW_L1_score','ATW_L2_score','SMW_score','SMW_L1_score','SMW_L2_score','DAW_score','FMW_score','EW_score','RAW_score','RAW_L1_score','RAW_L2_score','LRAW_score','LRAW_L1_score','LRAW_L2_score','RRAW_score','RRAW_L1_score','RRAW_L2_score','LROD_score','LROD_L1_score','LROD_L2_score','RROD_score','RROD_L1_score','RROD_L2_score'];
//Active cam setting - Live Monitor
$config['data']['camList'] = ['BSD_CAM1','DMS_CAM2','ADAS_CAM3'];
//Dangerous video filtering
$config['data']['dangerVideoFilter'] = "( CASE WHEN warn.danger IS NULL THEN ( CASE WHEN ((warn.wt = 2010001 AND CAST((warn.start_spd - warn.end_spd)/NULLIF(warn.duration,0) AS FLOAT) > 0.0015) OR (warn.wt = 2010003 AND warn.start_spd >= 60 AND warn.hw <= 4) OR (warn.wt = 2010011 AND warn.end_spd >= 20) OR (warn.wt = 2010012 AND warn.end_spd >= 20) OR (warn.wt = 2020001)) THEN 'Y' ELSE 'N' END ) ELSE warn.danger END )";

// Mailing setting (forget password)
$config['data']['mail_subject_zh-cn'] = '忘记密码 - GreenSafety';
$config['data']['mail_subject_zh-tw'] = '忘記密碼 - GreenSafety';
$config['data']['mail_subject_en'] = 'Forgot Password - GreenSafety';
$config['data']['mail_logo'] = __DIR__ . '/../config/mail_templates/logo-header.png';
$config['data']['mail_html_zh-cn'] = __DIR__ . '/../config/mail_templates/forgotpw-cn.html';
$config['data']['mail_html_zh-tw'] = __DIR__ . '/../config/mail_templates/forgotpw-tw.html';
$config['data']['mail_html_en'] = __DIR__ . '/../config/mail_templates/forgotpw-en.html';
$config['data']['mail_text_zh-cn'] = __DIR__ . '/../config/mail_templates/forgotpw-cn.txt';
$config['data']['mail_text_zh-tw'] = __DIR__ . '/../config/mail_templates/forgotpw-tw.txt';
$config['data']['mail_text_en'] = __DIR__ . '/../config/mail_templates/forgotpw-en.txt';

// Pre-login alert setting
/*$config['banner']['test']['default'] = __DIR__ . '/../config/banner_templates/test-en.html';
$config['banner']['test']['en'] = __DIR__ . '/../config/banner_templates/test-en.html';
$config['banner']['test']['zh-tw'] = __DIR__ . '/../config/banner_templates/test-tw.html';
$config['banner']['test']['zh-cn'] = __DIR__ . '/../config/banner_templates/test-cn.html';*/

// Reset Password path setting
$config['updatepwd']['uipath'] = 'updatePwd';
$config['resetpwd']['uipath'] = 'resetPwd';
$config['resetpwd']['linkpath'] = 'reset/password';

// Admin Panel setting
$config['adminpanel']['token'] = '350b1daa012d871cc3eac3aa1e8359bef8754214';

// Admin Panel FTP setting (driver card import)
$config['adminpanel']['drvcard_import']['host'] = '10.33.22.27';
$config['adminpanel']['drvcard_import']['hostprd'] = '10.33.22.27';
$config['adminpanel']['drvcard_import']['port'] = '21';
$config['adminpanel']['drvcard_import']['username'] = 'uploadftp';
$config['adminpanel']['drvcard_import']['password'] = 'XjxJa2wXyP2V8BwS';
$config['adminpanel']['drvcard_import']['ftpdir'] = 'Driver_Card_Import';
$config['adminpanel']['drvcard_import']['fulldir'] = 'C:\Users\Public\Documents\Driver_Card_Import';

// Admin Panel company management setting
$config['adminpanel']['company']['disclaimerdir'] = __DIR__ . '/../config/banner_templates';
?>
